src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedCaseOrderServiceTest.java

```diff
@@ -58,11 +83,109 @@ public void givenShouldNotPrintPackForApplicant_whenPrintAndMailGeneralOrderTrig
+        expectedBulkPrintDocuments.add("OtherHearingOrderDocumentsURL");
+
+        assertThat(bulkPrintArgumentCaptor.getAllValues().get(0).stream().map(
+            o -> o.getBinaryFileUrl()).collect(Collectors.toList()).containsAll(expectedBulkPrintDocuments),  is(true));
+    }
+
+    @Test
+    public void givenAllHearingDocumentsArePresentThenDoNotSendToBulkPrintWhenDigitalCase() throws JsonProcessingException {
+        CaseDetails caseDetails = buildHearingPackDocumentTestData();
+        caseDetails.getData().remove(PAPER_APPLICATION);
+
+        contestedCaseOrderService.printAndMailHearingDocuments(caseDetails, AUTH_TOKEN);
+
+        verifyNoInteractions(bulkPrintService);
+    }
+
+    @Test
+    public void latestDraftedHearingOrderDocumentIsNotAddedToPack() throws JsonProcessingException {
+        CaseDetails caseDetails = buildHearingPackDocumentTestData();
+        caseDetails.getData().remove(LATEST_DRAFT_HEARING_ORDER);
+
+        when(bulkPrintService.shouldPrintForApplicant(any())).thenReturn(true);
+        contestedCaseOrderService.printAndMailHearingDocuments(caseDetails, AUTH_TOKEN);
+
+        verify(bulkPrintService, times(1)).printApplicantDocuments(any(), any(), bulkPrintArgumentCaptor.capture());
+
+        List<String> expectedBulkPrintDocuments = new ArrayList<>();
+        expectedBulkPrintDocuments.add("AdditionalHearingDocumentURL");
+        expectedBulkPrintDocuments.add("OtherHearingOrderDocumentsURL");
+
+        assertThat(bulkPrintArgumentCaptor.getAllValues().get(0).stream().map(
+            o -> o.getBinaryFileUrl()).collect(Collectors.toList()).containsAll(expectedBulkPrintDocuments),  is(true));
+    }
+
+    @Test
+    public void latestAdditionalHearingDocumentIsNotAddedToPack() throws JsonProcessingException {
+        CaseDetails caseDetails = buildHearingPackDocumentTestData();
+        caseDetails.getData().remove(ADDITIONAL_HEARING_DOCUMENT_COLLECTION);
+
+        when(bulkPrintService.shouldPrintForApplicant(any())).thenReturn(true);
+        contestedCaseOrderService.printAndMailHearingDocuments(caseDetails, AUTH_TOKEN);
+
+        verify(bulkPrintService, times(1)).printApplicantDocuments(any(), any(), bulkPrintArgumentCaptor.capture());
+
+        List<String> expectedBulkPrintDocuments = new ArrayList<>();
+        expectedBulkPrintDocuments.add("HearingOrderBinaryURL");
+        expectedBulkPrintDocuments.add("OtherHearingOrderDocumentsURL");
+
+        assertThat(bulkPrintArgumentCaptor.getAllValues().get(0).stream().map(
+            o -> o.getBinaryFileUrl()).collect(Collectors.toList()).containsAll(expectedBulkPrintDocuments),  is(true));
+    }
+
+
+    private CaseDetails buildHearingPackDocumentTestData() throws JsonProcessingException {
+        callbackRequest = getContestedCallbackRequest();
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+
+        caseDetails.getData().put(PAPER_APPLICATION, "Yes");
```

Comment:
for future reference the only reason we needed to create all of this test data is because you call the documentHelper in you service. The documentHelper as I mention here https://tools.hmcts.net/confluence/pages/viewpage.action?pageId=1460549983#FRBestPractices-BaseServices>CommonFunctions can't be mocked out which means our tests also test the logic in the helper classes. In future if we move the methods to a base service and mock out the response then these tests become extremely simple